hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/webapp/FederationInterceptorREST.java

```diff
@@ -3371,17 +3372,19 @@ private <R> Map<SubClusterInfo, R> invokeConcurrent(Collection<SubClusterInfo> c
         }
 
         Exception exception = result.getException();
-
-        // If allowPartialResult=false, it means that if an exception occurs in a subCluster,
-        // an exception will be thrown directly.
-        if (!allowPartialResult && exception != null) {
+        if (exception != null) {
           throw exception;
         }
       } catch (Throwable e) {
         String subClusterId = subClusterInfo != null ?
             subClusterInfo.getSubClusterId().getId() : "UNKNOWN";
         LOG.error("SubCluster {} failed to {} report.", subClusterId, request.getMethodName(), e);
-        throw new YarnRuntimeException(e.getCause().getMessage(), e);
+        // If allowPartialResult=false, it means that if an exception occurs in a subCluster,
```

Comment:
After a careful read of the code, I have realized that allowPartialResult hasn't been effectively utilized due to its incorrect placement. We should evaluate whether to throw an exception in the catch block instead.